var/spack/repos/builtin/packages/py-nexusforge/package.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyNexusforge(PythonPackage):
+    """Blue Brain Nexus Forge is a domain-agnostic, generic and
+    extensible Python framework enabling non-expert users to create and
+    manage knowledge graphs.
+    """
+
+    homepage = "https://github.com/BlueBrain/nexus-forge"
+    pypi = "nexusforge/nexusforge-0.7.0.tar.gz"
+
+    version("0.8.0", sha256="4358505ead26e41c2a0c4e6113cf3a486c9661e2a3899394497a2b5a94b70424")
+    version("0.7.0", sha256="a8d2951d9ad18df9f2f4db31a4c18fcdd27bfcec929b03a3c91f133ea439413c")
+
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-hjson", type=("build", "run"))
+    depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-pandas", type=("build", "run"))
+    depends_on("py-nexus-sdk", type=("build", "run"))
+    depends_on("py-aiohttp", type=("build", "run"))
+    depends_on("py-rdflib@6.0.0:", type=("build", "run"))
+    depends_on("py-pyld", type=("build", "run"))
+    depends_on("py-pyshacl@0.17.2", type=("build", "run"))
+    depends_on("py-nest-asyncio@1.5.1:", type=("build", "run"))
+    depends_on("py-pyparsing@2.0.2:", type=("build", "run"))
+    depends_on("py-owlrl@5.2.3:", type=("build", "run"))
+    depends_on("py-elasticsearch-dsl@7.4.0", type=("build", "run"))
+    depends_on("py-scikit-learn", type="run")
```

Comment:
sklearn is optional, I would either remove it or add a variant (default to False) that enables it